examples/E2E-RAG-App/data/eval/eval.py

```diff
@@ -0,0 +1,271 @@
+        },
+        {
+            "Question": "What happens if an employee is absent without authorization for more than five consecutive working days?",
+            "Answer": "If an employee is absent from their job without proper authorization for more than five consecutive working days, or fails to return from an approved leave, their absence will be deemed an 'automatic resignation.' (Source: Page 19, 'Automatic Resignation' section)",
+        },
+        {
+            "Question": "How long is the normal probationary period for permanent civil service positions?",
+            "Answer": "The document states that all appointments to permanent civil service positions require a probationary period, but the duration is governed by the collective bargaining agreement. Absences from work will extend the probationary period. (Source: Page 14, 'Probationary Period' section)",
+        },
+        {
+            "Question": "What are employees required to do in case of a catastrophic event while off duty?",
+            "Answer": "Employees should ensure the safety of their family and follow their department's instructions. If phone lines are down, they are required to listen to the radio for any reporting instructions. (Source: Page 51, 'Catastrophic Event While off Duty' section)",
+        },
+        {
+            "Question": "What is the city's policy on accepting gifts from subordinates?",
+            "Answer": "Employees may not solicit or accept any gifts from any subordinate, or any candidate or applicant for a position as an employee or subordinate to them. (Source: Page 49, 'Gifts' section)",
+        },
+    ]
+
+    # Prepare lists to collect data
+    questions = []
+    generated_answers = []
+    retrieved_contexts = []
+    ground_truths = []
+
+    session_id = agent.create_session(f"session-{uuid.uuid4()}")
+    for qa in tqdm(qa_data, desc="Generating responses"):
+        question = qa["Question"]
+        ground_truth_answer = qa["Answer"]
+
+        cprint(f"Generating response for: {question}", "green")
+        try:
+            generated_answer, contexts = await get_response_with_context(
+                agent, question, session_id
+            )
+            cprint(f"Response: {generated_answer}", "green")
+
+            # Append data to lists
+            questions.append(question)
+            generated_answers.append(generated_answer)
+            retrieved_contexts.append(contexts)
+            ground_truths.append(ground_truth_answer)
+        except Exception as e:
+            cprint(f"Error generating response for {question}: {e}", "red")
+
+    # Create a Dataset for RAGAS
+    eval_data = Dataset.from_dict(
+        {
+            "user_input": questions,
+            "response": generated_answers,
+            "retrieved_contexts": retrieved_contexts,
+            "reference": ground_truths,
+        }
+    )
+
+    # Run evaluation
+    result = evaluate(
```

Comment:
We have native /evals API with Llama Stack now. Could we try evaluate the generated datasets with Llama Stack /evals & /scoring? Example here: https://github.com/meta-llama/llama-stack-apps/pull/117

Reply:
That's amazing! will try it out

Reply:
That's amazing! will take a look and try it out